src/server/scripts/Spells/spell_dk.cpp

```diff
@@ -1689,6 +1655,73 @@ class spell_dk_death_grip_initial : public SpellScriptLoader
     }
 };
 
+// 49560 - Death Grip (second part)
+class spell_dk_death_grip : public SpellScriptLoader
+{
+public:
+    spell_dk_death_grip() : SpellScriptLoader("spell_dk_death_grip") { }
+
+    class spell_dk_death_grip_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_dk_death_grip_SpellScript);
+
+        void HandleDummy(SpellEffIndex /*effIndex*/)
+        {
+            int32 damage = GetEffectValue();
+            Position const* pos = GetExplTargetDest();
+            if (Unit* target = GetHitUnit())
+            {
+                if (!target->HasAuraType(SPELL_AURA_DEFLECT_SPELLS)) // Deterrence
+                    target->CastSpell(pos->GetPositionX(), pos->GetPositionY(), pos->GetPositionZ(), damage, true);
+            }
+        }
+
+        void Register() override
+        {
+            OnEffectHitTarget += SpellEffectFn(spell_dk_death_grip_SpellScript::HandleDummy, EFFECT_0, SPELL_EFFECT_DUMMY);
+        }
+
+    };
+
+    SpellScript* GetSpellScript() const override
+    {
+        return new spell_dk_death_grip_SpellScript();
+    }
+};
+
+// 49575 - Death Grip (third part. pull effect only)
+class spell_dk_death_grip_pull : public SpellScriptLoader
+{
+public:
+    spell_dk_death_grip_pull() : SpellScriptLoader("spell_dk_death_grip_pull") { }
+
+    class spell_dk_death_grip_pull_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_dk_death_grip_pull_SpellScript);
+
+        SpellCastResult CheckCast()
+        {
+            Creature* creatureTarget = GetCaster()->ToCreature(); // the caster is the unit affected by Death Grip (see spell script of previous spell)
+            if (creatureTarget && creatureTarget->GetCreatureTemplate()->flags_extra & CREATURE_FLAG_EXTRA_NO_PULL_BY_GRIP)
```

Comment:
@ccrs I want the caster to be notified that the target is immuned to one of the effects of the spell (in this case, the pull component). When this method is reached, it's already too late to send that "immune" message.

@DDuarte Sure but I don't know of any other spell that could make use of that flag actually. Do you have any in mind?